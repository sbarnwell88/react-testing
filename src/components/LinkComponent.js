import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";


const STATUS = {
    HOVERED: 'hovered',
    NORMAL: 'normal'
}

class LinkComponent extends Component {
    constructor() {
        super();

        this._onMouseEnter = this._onMouseEnter.bind(this);
        this._onMouseLeave = this._onMouseLeave.bind(this);

        this.state = {
            class: STATUS.NORMAL
        }
    }

    _onMouseEnter() {
        this.setState({class: STATUS.HOVERED});
    }
    
    _onMouseLeave() {
        this.setState({class: STATUS.NORMAL});
    }
    render() {
        return (
            <div>
                Hello from Link
                <Link to={'/'}
                    className={this.state.class}
                    href={this.props.page || '#'}
                    onMouseEnter={this._onMouseEnter}
                    onMouseLeave={this._onMouseLeave}
                >
                    {this.props.children}
                
                Home</Link>
            </div>
        );
    }
}

export default LinkComponent;