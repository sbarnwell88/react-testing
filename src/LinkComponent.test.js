import React from 'react';
import LinkComponent from './components/LinkComponent';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Link } from "react-router-dom";


it('Link changes state when hovered', () => {
    const component = renderer.create(
        <Router>
        <Link to='http://localhost:3000/link'>Home</Link>
        </Router>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  
    // manually trigger the callback
    tree.props.onMouseEnter;
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  
    // manually trigger the callback
    tree.props.onMouseLeave;
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

  });
  