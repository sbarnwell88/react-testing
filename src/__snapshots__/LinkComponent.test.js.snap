// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes state when hovered 1`] = `
<a
  href="http://localhost:3000/link"
  onClick={[Function]}
>
  Home
</a>
`;

exports[`Link changes state when hovered 2`] = `
<a
  href="http://localhost:3000/link"
  onClick={[Function]}
>
  Home
</a>
`;

exports[`Link changes state when hovered 3`] = `
<a
  href="http://localhost:3000/link"
  onClick={[Function]}
>
  Home
</a>
`;
